dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EmailPropertyFilter.java

```diff
@@ -0,0 +1,43 @@
+package io.dockstore.webservice.helpers;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.BeanPropertyWriter;
+import com.fasterxml.jackson.databind.ser.PropertyWriter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import io.dockstore.webservice.core.User.Profile;
+import org.apache.commons.validator.routines.EmailValidator;
+
+/**
+ * Conditionally removes Strings that look like email addresses
+ * Example from https://www.baeldung.com/jackson-serialize-field-custom-criteria
+ */
+public class EmailPropertyFilter extends SimpleBeanPropertyFilter {
+    EmailValidator emailValidator = EmailValidator.getInstance();
+
+    @Override
+    public void serializeAsField(Object pojo, JsonGenerator jgen, SerializerProvider provider, PropertyWriter writer) throws Exception {
+        if (include(writer)) {
+            if (!writer.getName().equals("username")) {
+                writer.serializeAsField(pojo, jgen, provider);
+                return;
+            }
+            Profile profile = (Profile) pojo;
+
+            if (!emailValidator.isValid(profile.username)) {
```

Comment:
If this condition is true, nothing will get serialized. Is that the intent? It could be, but then there is an explicit `serializeAsOmittedField` on line 31 which is making me ask this question. I haven't used `PropertyWriter`, so this may be good, but just double-checking.

Reply:
Yes,
I haven't used it either before, but looking at the Javadoc it looks like this is for stuff like tsv or csv where you cannot outright skip a field (for an arbitrary row) like with json